spaceR/spaceR/ViewModels/SignupViewModel.swift

```diff
@@ -10,42 +10,61 @@ import UIKit
 
 class SignupViewModel {
     
-    private var userFirstName = ""
-    private var userLastName = ""
-    private var userEmail = ""
-    private var userPassword = ""
+    private var userFirstName: String
+    private var userLastName: String
+    private var userEmail: String
+    private var userPassword: String
+    private var validCredentials: Bool
+    private var saveStatus: Bool
+    private weak var delegate: ViewModelDelegate?
+    private var repository: SignupRepositoryType?
     
-    private let coreDataPersistantObject = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+    private let coreDataPersistantObject = (UIApplication.shared.delegate
+                                            as? AppDelegate)?.persistentContainer.viewContext
     
-    func set(firstName: String) {
-        userFirstName = firstName
+    init(repository: SignupRepositoryType,
+         delegate: ViewModelDelegate) {
+        self.delegate = delegate
+        self.repository = repository
+        self.userFirstName = ""
```

Comment:
Creating a Model for all these variables would be neater 😄

Reply:
so i should create a model for my model?😅
(if so, i will do it in the next pr)

Reply:
So remember this is a `ViewModel` a `model` holds data for us, it is usually a struct.